src/components/EtherspotTransactionKit.tsx

```diff
@@ -10,34 +11,67 @@ import ProviderWalletContextProvider from '../providers/ProviderWalletContextPro
 import EtherspotContextProvider from '../providers/EtherspotContextProvider';
 
 interface EtherspotTransactionKitProps extends React.PropsWithChildren {
-  provider: WalletProviderLike;
+  provider: WalletProviderLike | WalletProviderLikeModular;
   chainId?: number;
   accountTemplate?: AccountTemplate;
   dataApiKey?: string;
   bundlerApiKey?: string;
+  modular?: boolean;
 }
 
 const EtherspotTransactionKit = ({
   children,
   provider,
   chainId = 1,
-  accountTemplate = Factory.ETHERSPOT,
+  accountTemplate,
   dataApiKey,
   bundlerApiKey,
-}: EtherspotTransactionKitProps) => (
+  modular = true,
+}: EtherspotTransactionKitProps) => {
+  let accountTemp;
+
+  if (accountTemplate) {
+    switch (accountTemplate) {
+      case 'zeroDev':
+        if (!modular) {
+          accountTemp = Factory.ZERO_DEV;
+        } else {
+          console.warn('You cannot use a ZeroDev Account template with the modular functionality.');
+        }
+        break;
+      case 'simpleAccount':
+        if (!modular) {
+          accountTemp = Factory.SIMPLE_ACCOUNT;
+        } else {
+          console.warn('You cannot use a Simple Account template with the modular functionality.');
+        }
+        break;
+      case 'etherspot':
+        accountTemp = modular ? ModularFactory.ETHERSPOT : Factory.ETHERSPOT;
+        break;
```

Comment:
We would only need another `switch` / `case` here for `etherspotModular` - no if conditions needed within the other cases

Reply:
I have updated this